src/HDF5.jl

```diff
@@ -157,27 +153,28 @@ function Base.setindex!(parent::Union{File,Group}, val, path::Union{AbstractStri
     write(parent, path, val; pv...)
 end
 
-
 # end of high-level interface
 
-
 include("api_midlevel.jl")
 
-
 #API.h5s_get_simple_extent_ndims(space_id::API.hid_t) = API.h5s_get_simple_extent_ndims(space_id, C_NULL, C_NULL)
 
-
 # Functions that require special handling
 
 const libversion = API.h5_get_libversion()
 
 ### Property manipulation ###
-get_access_properties(d::Dataset)   = DatasetAccessProperties(API.h5d_get_access_plist(d))
-get_access_properties(f::File)      = FileAccessProperties(API.h5f_get_access_plist(f))
-get_create_properties(d::Dataset)   = DatasetCreateProperties(API.h5d_get_create_plist(d))
-get_create_properties(g::Group)     = GroupCreateProperties(API.h5g_get_create_plist(g))
-get_create_properties(f::File)      = FileCreateProperties(API.h5f_get_create_plist(f))
-get_create_properties(a::Attribute) = AttributeCreateProperties(API.h5a_get_create_plist(a))
+function get_context_property(name::Symbol)
+    local_context = get(task_local_storage(), :hdf5_context, CONTEXT)
+    property = getfield(local_context, name)
+    isvalid(property) ? property : nothing
+end
+get_access_properties(d::Dataset)   = something(get_context_property(:dataset_access), DatasetAccessProperties(API.h5d_get_access_plist(d)))
+get_access_properties(f::File)      = something(get_context_property(:file_access), FileAccessProperties(API.h5f_get_access_plist(f)))
+get_create_properties(d::Dataset)   = something(get_context_property(:dataset_create), DatasetCreateProperties(API.h5d_get_create_plist(d)))
+get_create_properties(g::Group)     = something(get_context_property(:group_create), GroupCreateProperties(API.h5g_get_create_plist(g)))
+get_create_properties(f::File)      = something(get_context_property(:file_create), FileCreateProperties(API.h5f_get_create_plist(f)))
+get_create_properties(a::Attribute) = something(get_context_property(:attribute_create), AttributeCreateProperties(API.h5a_get_create_plist(a)))
```

Comment:
```suggestion
get_access_properties(d::Dataset)   = DatasetAccessProperties(API.h5d_get_access_plist(d))
get_access_properties(f::File)      = FileAccessProperties(API.h5f_get_access_plist(f))
get_create_properties(d::Dataset)   = DatasetCreateProperties(API.h5d_get_create_plist(d))
get_create_properties(g::Group)     = GroupCreateProperties(API.h5g_get_create_plist(g))
get_create_properties(f::File)      = FileCreateProperties(API.h5f_get_create_plist(f))
get_create_properties(a::Attribute) = AttributeCreateProperties(API.h5a_get_create_plist(a))
```

Revert this. Just call `get_context_property` directly if we are interested in getting a property from the context. If we want to choose between object and context properties, that should be further downstream on a per-property basis.

Reply:
This has not been resolved...